internal/app/machined/pkg/system/services/kubelet.go

```diff
@@ -386,6 +402,100 @@ func writeKubeletConfig(r runtime.Runtime) error {
 	return ioutil.WriteFile("/etc/kubernetes/kubelet.yaml", buf.Bytes(), 0o600)
 }
 
+//nolint:gocyclo
+func writeKubeletCerts(r runtime.Runtime) error {
+	nodename, err := r.NodeName()
+	if err != nil {
+		return err
+	}
+
+	var (
+		pkiLink string
+		pkiName string
+		pki     []byte
+	)
+
+	ts := time.Now().Format("2006-01-02-15-04-05")
+
+	ca, err := x509.NewCertificateAuthorityFromCertificateAndKey(r.Config().Cluster().CA())
+	if err != nil {
+		return fmt.Errorf("failed to parse CA certificate: %w", err)
+	}
+
+	kubeletClientKeyPair, err := x509.NewKeyPair(ca,
+		x509.CommonName(constants.KubernetesKubeletCommonNamePrefix+nodename),
```

Comment:
evaluating certificates using Common Names is common... but discouraged. And some go libraries will actually migrate away from doing that. I suggest you instead rely on setting a SubjectAltName instead to avoid future problems.